src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -275,28 +290,14 @@ void VideoDecoder::initializeDecoder() {
 }
 
 std::unique_ptr<VideoDecoder> VideoDecoder::createFromFilePath(
-    const std::string& videoFilePath,
-    const VideoDecoder::DecoderOptions& options) {
-  AVInput input = createAVFormatContextFromFilePath(videoFilePath);
-  std::unique_ptr<VideoDecoder> decoder(new VideoDecoder());
-  decoder->formatContext_ = std::move(input.formatContext);
-  decoder->options_ = options;
-  decoder->initializeDecoder();
-  return decoder;
+    const std::string& videoFilePath) {
+  return std::unique_ptr<VideoDecoder>(new VideoDecoder(videoFilePath));
 }
 
 std::unique_ptr<VideoDecoder> VideoDecoder::createFromBuffer(
     const void* buffer,
-    size_t length,
-    const VideoDecoder::DecoderOptions& options) {
-  TORCH_CHECK(buffer != nullptr, "Video buffer cannot be nullptr!");
```

Comment:
Note that this `TORCH_CHECK` was almost certainly optimized out in optimized builds. Because we already used `buffer`, it being null at this point would be undefined behavior, and the C++ compiler is allowed to assume it won't happen. Null checks must always be before use.